component/azstorage/utils.go

```diff
@@ -354,6 +354,21 @@ func storeDatalakeErrToErr(err error) uint16 {
 	return ErrNoErr
 }
 
+// Convert datalake storage error to common errors
+func storeFileErrToErr(err error) uint16 {
+	if serr, ok := err.(azfile.StorageError); ok {
+		switch serr.ServiceCode() {
+		case azfile.ServiceCodeShareAlreadyExists:
+			return ErrFileAlreadyExists
+		case azfile.ServiceCodeShareNotFound:
```

Comment:
I think you need to be matching these errors with the FileAlreadyExists/DirectoryAlreadyExists, NotFound ones too